lib/split/experiment.rb

```diff
@@ -127,16 +128,30 @@ def resettable=(resettable)
     def alternatives=(alts)
       @alternatives = alts.map do |alternative|
         if alternative.kind_of?(Split::Alternative)
+          alternative.version = version
           alternative
         else
-          Split::Alternative.new(alternative, @name)
+          Split::Alternative.new(alternative, @name, version)
+        end
+      end
+    end
+
+    def alternatives(version_key = version)
+      if version_key == version && @alternatives.map(&:version).uniq.first == version
```

Comment:
Something about this gives me pause, the second half of the `&&` clause specifically. If I'm understanding it, it's check for an Experiment with version Y, but a set of Alternatives with a different version.  Is that expected to happen?  It feels like something's very wrong at that point, possible a bug? Or am I missing something?

Reply:
This is a bit of a fudge so that it can get around reinstanciating alternatives if it finds keys for them already. Will investigate further.

Reply:
What was the outcome?

Reply:
Yep, confirmed that it's to prevent a new alternative from being initialized because it would reset the counters. Alternatives are were the trial counts and completions are kept.